addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/FetchMetadataRequestHeadersScanRule.java

```diff
@@ -0,0 +1,267 @@
+import org.zaproxy.addon.commonlib.CommonAlertTag;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+public class FetchMetadataRequestHeadersScanRule extends PluginPassiveScanner {
+    private static final String FETCH_METADATA_REQUEST_MESSAGE_PREFIX =
+            "pscanalpha.metadata-request-headers.";
+    private static final int PLUGIN_ID = 90005;
+    private final List<FetchMetaDataRequestHeaders> rules =
+            Arrays.asList(
+                    new SecFetchSite(this::newAlert),
+                    new SecFetchMode(this::newAlert),
+                    new SecFetchDest(this::newAlert),
+                    new SecFetchUser(this::newAlert));
+
+    private static final Map<String, String> ALERT_TAGS =
+            CommonAlertTag.toMap(CommonAlertTag.WSTG_V42_SESS_05_CSRF);
+
+    @Override
+    public void scanHttpRequestSend(HttpMessage msg, int id) {
+        rules.forEach(
+                rule ->
+                        rule.build(msg.getRequestHeader(), rule.getHeader(), rule.getValidValues())
+                                .forEach(AlertBuilder::raise));
+    }
+
+    @Override
+    public int getPluginId() {
+        return PLUGIN_ID;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(FETCH_METADATA_REQUEST_MESSAGE_PREFIX + "name");
+    }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        List<Alert> alerts = new ArrayList<>();
+
+        alerts.addAll(
+                rules.stream()
+                        .map(s -> s.alert("missing.", "").build())
+                        .collect(Collectors.toList()));
+        alerts.addAll(
+                rules.stream()
+                        .map(s -> s.alert("invalid-values.", "").build())
+                        .collect(Collectors.toList()));
+
+        return alerts;
+    }
+
+    @Override
+    public Map<String, String> getAlertTags() {
+        return ALERT_TAGS;
+    }
+
+    abstract static class FetchMetaDataRequestHeaders {
+        private final Supplier<AlertBuilder> newAlert;
+        private final String alertRef;
+
+        FetchMetaDataRequestHeaders(Supplier<AlertBuilder> newAlert, String alertReference) {
+            this.newAlert = newAlert;
+            this.alertRef = PLUGIN_ID + alertReference;
+        }
+
+        protected abstract String getHeader();
+
+        protected abstract String getString(String param);
+
+        protected abstract List<String> getValidValues();
+
+        List<AlertBuilder> build(
+                HttpRequestHeader httpRequestHeaders, String header, List<String> valid_values) {
+            List<AlertBuilder> alert = new ArrayList<>();
+            List<HttpHeaderField> httpHeaderFields = httpRequestHeaders.getHeaders();
+
+            boolean hasHeader =
+                    httpHeaderFields.stream()
+                            .anyMatch(httpHeaderField -> httpHeaderField.getName().equals(header));
```

Comment:
Should this be using `equalsIgnoreCase`?

Reply:
Yes, just realized headers are case-insensitive. Thanks for pointing it.

Reply:
done